### PR TITLE
retain-original-class-as-interface: rename interface as RateLimiter t…

### PR DESCRIPTION
**Pull request by gjesse**
…o limit migration pain

### CODE REVIEW

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java [hunk 1]
```diff
@@ -341,14 +341,14 @@ private double getMeanRate(double rate, long startTime) {
     }
 
 
-    private TokenBucketRateLimiter buildRateLimiter(LongSupplier clock, int maxTokens, int requestRate, Duration refillPeriod) {
+    private TokenBucketRateLimiter buildRateLimiter(LongSupplier clock, int maxTokens, int refillValue, Duration refillPeriod) {
```
Comment (davewingate) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java]: +1 for this name change .... much clearer to me

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/StrictRateLimiter.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.exceptions.JedisDataException;
+
+import java.util.OptionalLong;
+
+/**
+ * Rate limiter implementation that will never allow
+ * greater than the configured requests/second
+ */
+public class StrictRateLimiter implements RateLimiter {
```
Comment (davewingate) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/StrictRateLimiter.java]: I'm assuming that you're just renaming this impl class from ```RateLimiter``` to ```StrictRateLimiter``` and not changing any of the impl details, right?

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java [hunk 3]
```diff
@@ -251,12 +252,11 @@ public void testExpire() {
...
 
         assertFalse(tokenBucketRateLimiter.getWaitTimeForPermit().isPresent());
-        // creates an entry that should expire in 40ms
+        // creates an entry that should expire in 2s
         try (Handle handle = rdbi.open()) {
             final Jedis jedis = handle.jedis();
             assertEquals(jedis.ttl(tokenBucketRateLimiter.getKey()).longValue(), 2L);
-            Uninterruptibles.sleepUninterruptibly(2, TimeUnit.SECONDS);
-            assertFalse(jedis.exists(tokenBucketRateLimiter.getKey()));
+            await().atMost(5, TimeUnit.SECONDS).until(() -> !jedis.exists(tokenBucketRateLimiter.getKey()));
```
Comment (davewingate) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java]: hawt ... this is new to me ... seems super useful

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/StrictRateLimiter.java (hunk 2 above)
Comment (gjesse) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/StrictRateLimiter.java]: correct

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java (hunk 3 above)
Comment (gjesse) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/ratelimiter/TokenBucketRateLimiterTest.java]: yeah it is da bomb when testing async code

**Comment by gjesse** [comment]
That's a good point - i had forgotten that this is a public repo.  Unfortunately, I already broke the public api with the previous PR. This one gets it closer to the original but it's still changed from what it was originally. if we are going to maintain full support for the original api then more work needs to be done.

@phutwo @matthewbogner , thoughts?

**Comment by matthewbogner** [comment]
Regarding the API change.... we need to strive to maintain that backward compatibility as much as reasonably possible.  Since any community engagement is pretty low to this point, I think we can make the API change.  If this OSS starts getting a lot more traction (which will be evidenced by issues and PRs coming from non-lithium employees) then we should start being more strict about semantic versioning.

**Comment by gjesse** [comment]
word - i think also as an 0.0.xxx version still we have some latitude.

**Pull request merged**